delayed-state-rendering.md

```diff
@@ -0,0 +1,273 @@
+- Feature Name: Delayed State Rendering
+- Start Date: 2019-06-21
+- SEP Status: Draft
+- SEP PR: (leave this empty)
+- Salt Issue: (leave this empty)
+- Author: Joseph Nix, nixjdm at terminallabs dot com, [nixjdm@GitHub](https://github.com/nixjdm)
+
+# Summary
+
+This SEP proposes a way of delaying rendering of portions of states until after other states have executed.
+
+A fundamental aspect of Salt is the ability to come to a desired endstate by executing a single highstate. This is currently limited when encountering a particular, fairly common circumstance. Consider a state that is truly stateful, but has an unpredictable effect. That is, this state is idempotent, but you can't know ahead of time a particular effect it has - like a hash. If you want to write a second state that needs this unpredictable information at its render time, then these two states cannot exist in the highstate together. This is because the rendering of the entire highstate happens prior to the execution of any of the states, i.e. the second state needs the first to complete before its render.
+
+Currently, to get around this issue, I often write orchestration states. Other Salt tools like beacons and reactors can be used too. These tools have great value, but when the desired outcome is just part of the idempotent desired state of a minion, I really just want them in a highstate. What I propose adds complexity to the highstate, but I think it's a worthy trade to get rid of the complexity of these workarounds. At least, the choice would then be left to the developer to use delayed rendering, or use an alternate solution like orchestration.
```

Comment:
Started talking with I's in this paragraph. suggest to replace with 'we' and 'you'.

Reply:
Right, I thought I was being inconsistent there. I was going to convert to straight "I"s. Maybe I'll switch to we.